components/link/link.js

```diff
@@ -74,10 +75,10 @@ class Link extends FocusMixin(LitElement) {
 			 */
 			main: { type: Boolean, reflect: true },
 			/**
-			 * Whether to truncate the link with ellipsis
-			 * @type {boolean}
+			 * The number of lines to display before truncating text with an ellipsis. The text will not be truncated unless a value is specified.
+			 * @type {number}
 			 */
-			overflowEllipsis: { type: Boolean, attribute: 'overflow-ellipsis', reflect: true },
+			lines: { type: Number },
```

Comment:
Should default it to `0` in the constructor. (collapsible-panel-summary-item should probably be doing that too)